onnx_chainer/export.py

```diff
@@ -103,11 +103,28 @@ def rename_variable_name(
         context.set_name(variables, new_name, pinned=True)
 
 
+class InputCacheHook(chainer.FunctionHook):
+    """cache raw inputs
+
+    Function nodes manage inputs variable nodes using weak reference. When
+    variable is made as temporary value, exporter cannot get the corresponded
+    variable from the variable node because the reference is collected. To
+    resolve it, cache all inputs and will use when make computational graph.
+    """
+
+    def __init__(self):
+        self.func_inputs = {}
+
+    def forward_postprocess(self, fn, in_data):
+        # in_data is ndarray, not variable
+        self.func_inputs[id(fn)] = in_data
```

Comment:
I think this will increase memory consumption. I think what we need to keep here is `in_data` which is not wrapped by `chainer.Variable` in `inputs` of `FunctionNode.forward`. However, it seems to be impossible to check this?

Reply:
Exactly! I tried `LinkHook` instead.